cli/azd/pkg/project/container_helper.go

```diff
@@ -201,7 +204,27 @@ func (ch *ContainerHelper) Credentials(
 		return nil, err
 	}
 
-	return ch.containerRegistryService.Credentials(ctx, targetResource.SubscriptionId(), loginServer)
+	var credential *azcli.DockerCredentials
+	credentialsError := retry.Do(
+		ctx,
+		retry.WithMaxDuration(90*time.Second, retry.NewExponential(1*time.Second)),
+		func(ctx context.Context) error {
+			cred, err := ch.containerRegistryService.Credentials(ctx, targetResource.SubscriptionId(), loginServer)
+			if err != nil {
+				var httpErr *azcore.ResponseError
+				if errors.As(err, &httpErr) {
+					if httpErr.StatusCode == 404 {
+						// Retry if the registry is not found while logging in
+						return retry.RetryableError(err)
+					}
+				}
+				return err
+			}
+			credential = cred
+			return nil
+		})
+
+	return credential, credentialsError
```

Comment:
I'd recommend we move this retry down into the `containerRegistryService.Credentials` function since that is our lowest level component so all callers can take advantage. ex) `containerRegistryService.Login` also calls into this `Credentials()` function.

Reply:
`containerRegistryService` is an interface.  Applying the `retry` to `.Credentials` implementation will not fix the issue if there's another implementation added sometime in the future.

I was actually going to use the interface to mock the 404 response :)

I am not sure why we are duplicating the logic for `login`, from Credentials and from Login.  Probably that requires another change to consolidate both APIs into just one...  The `Credentials` API seems unique to service-targe dotnet-appcontainer.